test/suites/integration/subnet_test.go

```diff
@@ -134,6 +136,24 @@ var _ = Describe("Subnets", func() {
 			lo.Map(subnets[shuffledAZs[0]], func(subnetID string, _ int) types.GomegaMatcher { return HaveValue(Equal(subnetID)) })...,
 		)))
 	})
+
+	It("should have the AWSNodeTemplateStatus for subnets", func() {
+		subnets := getSubnetList(map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName})
+		Expect(len(subnets)).ToNot(Equal(0))
+
+		provider := awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{
+			AWS: v1alpha1.AWS{
+				SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+				SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			},
+		})
+
+		env.ExpectCreated(provider)
+
+		var ant v1alpha1.AWSNodeTemplate
+		Expect(env.Client.Get(env, client.ObjectKeyFromObject(provider), &ant)).To(Succeed())
```

Comment:
If an `ExpectExists` function doesn't already exist, can we move a function that does this check and retrieval as an expectation function?